[DependencyInjection] [Autowiring] Allow undefined nullable parameters (6.3)

radar3301 opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | yes
| New feature?  | yes <!-- please update src/**/CHANGELOG.md files -->
| Deprecations? | no <!-- please update UPGRADE-*.md and src/**/CHANGELOG.md files -->
| Tickets       | Fix #50020 <!-- prefix each issue number with "Fix #", no need to create an issue if none exists, explain below instead -->
| License       | MIT
| Related       | PR #50022

Allows autowiring of undefined nullable parameters, similar to existing behavior for non-parameter values. See related issue for more detail.

radar3301 committed:
Allow undefined nullable parameters

radar3301 committed:
Update AutowirePass.php

radar3301 commented:
Grr... Tried renaming local patch to https://github.com/radar3301/symfony/tree/autowire-undefined-nullable-6.3

nicolas-grekas commented:
Thanks for the PRs. Usually we ask for the patch on the lowest affected branch, and we merge up periodically. When the merge generates conflicts, we fix the conflict and we sometimes ask the author for their help if it's non trivial. I'm therefor closing this PR but it'll definitely help when merging #50022, if accepted!

radar3301 commented:
> Thanks for the PRs. Usually we ask for the patch on the lowest affected branch, and we merge up periodically. When the merge generates conflicts, we fix the conflict and we sometimes ask the author for their help if it's non trivial. I'm therefor closing this PR but it'll definitely help when merging #50022, if accepted!

Hey Nicolas,

The Autowire Attribute was added in 6.1, and handling of the Autowire attribute was added directly to `autowireMethod`.
6.2 added a `processAttribute` method to do the actual processing, and rewrote `autowireMethod` to handle the attribute.
Then 6.3 removed `processAttribute` and moved the attribute handling back into `autowireMethod`.

It's all a big mess, hence the 3 different patches I created.
